src/components/ec-metroline-item/ec-metroline-item.vue

```diff
@@ -103,69 +103,62 @@
   </div>
 </template>
 
-<script>
+<script setup>
+import {
+  computed, inject, onBeforeUnmount, onMounted, toRefs,
+} from 'vue';
+
 import * as MetrolineItemStatus from '../../enums/metroline-item-status';
 import EcIcon from '../ec-icon';
 
-export default {
-  name: 'EcMetrolineItem',
-  components: { EcIcon },
-  inject: ['metroline'],
-  props: {
-    id: {
-      type: Number,
-      required: true,
-    },
-    badgeText: {
-      type: String,
-    },
-  },
-  computed: {
-    isReadOnly() {
-      return this.metroline.isCompleted;
-    },
-    isLast() {
-      return this.id === this.metroline.lastItemId;
-    },
-    status() {
-      if (this.id < this.metroline.activeItemId || this.metroline.isCompleted) {
-        return MetrolineItemStatus.COMPLETED;
-      }
-
-      if (this.id === this.metroline.activeItemId) {
-        return MetrolineItemStatus.ACTIVE;
-      }
-
-      return MetrolineItemStatus.NEXT;
-    },
-    isNext() {
-      return this.status === MetrolineItemStatus.NEXT;
-    },
-    isActive() {
-      return this.status === MetrolineItemStatus.ACTIVE;
-    },
-    isCompleted() {
-      return this.status === MetrolineItemStatus.COMPLETED;
-    },
-  },
-  created() {
-    this.metroline.register(this.id);
-  },
-  beforeUnmount() {
-    this.metroline.unregister(this.id);
+const metroline = inject('metroline');
+
+const props = defineProps({
+  id: {
+    type: Number,
+    required: true,
   },
-  methods: {
-    goToNext() {
-      this.metroline.goToNext(this.id);
-    },
-    activateItem() {
-      this.metroline.goTo(this.id);
-    },
-    complete() {
-      this.metroline.complete();
-    },
+  badgeText: {
+    type: String,
   },
-};
+});
+
+const { id } = toRefs(props);
```

Comment:
You don't have to destructure the props, you can also use `props.id` directly. It shows the origin of the prop and you don't need `.value` then.